src/sas/qtgui/Plotting/Slicers/BaseInteractor.py

```diff
@@ -139,7 +141,8 @@ def onDrag(self, ev):
             self.clickx, self.clicky = ev.xdata, ev.ydata
             self.move(ev.xdata, ev.ydata, ev)
         else:
-            self.restore(ev)
+            logging.warning("the handle cannot be moved outside the data")
+            #self.restore(ev)
```

Comment:
Three reasons.
  * The primary reason is that it is completely unnecessary. since the illegal move is caught BEFORE it actually moves, there is nothing to restore. The cascade of changes has not taken place and **all** the elements are therefore in a perfectly legal state.  NOTE: I removed quite a few of those instances, many of which were introduced early in this branch to "copy" the existing structure.
  *  Secondly, jumping from what is a totally legal state all the way back to when the mouse click was first engaged can be annoying depending on how much effort the user is taking in moving the mouse "just so" (and again ... did I mention "unnecessary"
  * It makes the code cleaner and easier to follow. Originally (back in the dawn of time when slicers were first introduced) all save and restores were done by the `BaseInteractor`. That is no longer the case. Besides making it unnecessarily complicated to find all the occurrences of `restore` there were edge cases when you ended up with 2 restores or a restore when it really should not have restored.

Reply:
comment now removed